### PR TITLE
56 crptg1 test node fixing

### PR DESCRIPTION
**Pull request by gfigneczi1**

### CODE REVIEW

File: crp_APL/test/include/testNode/testNode.hpp [hunk 1]
```diff
@@ -37,32 +41,59 @@ class TestNode : public rclcpp::Node
...
 
     rclcpp::TimerBase::SharedPtr m_timer_;
 
     // target values coming from the vehicle
-    float m_roadWheelAngleTarget{0.0f};
-    float m_vehicleSpeedTarget{0.0f};
-    float p_wheelBase{2.7f}; // in meters
+    double m_roadWheelAngleTarget{0.0f};
+    double m_vehicleSpeedTarget{0.0f};
+    double p_wheelBase{2.7f}; // in meters
```
Comment (mesmatyi) [inline crp_APL/test/include/testNode/testNode.hpp:81]: These are fixed or overwritten in runtime ?

File: crp_APL/test/include/testNode/testNode.hpp [hunk 2]
```diff
@@ -37,32 +41,59 @@ class TestNode : public rclcpp::Node
...
 
-    // vehicle model, state variables: [X Y Theta], input variable: road wheel angle, parameter: longitudinal speed
-    float x[3]{0.0f};
+    double m_maximumSpeedInit{20.0f}; // in mps
+    
+    double p_previewDistance = 100; // in meters
+    double m_maximumSpeed{0.0f};
+    bool m_noEnoughPointsLeft{false};
```
Comment (mesmatyi) [inline crp_APL/test/include/testNode/testNode.hpp:87]: maybe these params can be in a launch file ?

File: crp_APL/test/src/testNode.cpp [hunk 3]
```diff
@@ -93,43 +103,99 @@ float crp::apl::TestNode::getYawFromQuaternion(const geometry_msgs::msg::Quatern
...
-    m_egoMsg.pose.pose.orientation = tf2::toMsg(tf2::Quaternion(tf2::Vector3(0, 0, 1), x[2]));
-
-    m_egoMsg.twist.twist.linear.x = m_vehicleSpeedTarget;
+    m_kinematicState.pose_with_covariance.pose.orientation = 
+        tf2::toMsg(tf2::Quaternion(tf2::Vector3(0, 0, 1), x[2])); 
 
-    m_egoMsg.tire_angle_front = m_roadWheelAngleTarget;
+    m_kinematicState.twist_with_covariance.twist.linear.x = m_vehicleSpeedTarget;
 
     return;
```
Comment (mesmatyi) [inline crp_APL/test/src/testNode.cpp:200]: Maybe lateral acceleration can be publiseh too, or would be neat to have it in the future, for other types of controllers

**Review by mesmatyi — COMMENTED** [review]
Left a couple of comments none of it is a must have...

File: crp_APL/test/include/testNode/testNode.hpp (hunk 2 above)
Comment (gfigneczi1) [reply crp_APL/test/include/testNode/testNode.hpp:87]: done

File: crp_APL/test/src/testNode.cpp (hunk 3 above)
Comment (gfigneczi1) [reply crp_APL/test/src/testNode.cpp:200]: done

**Pull request merged**